crates/llms/tests/llms/create.rs

```diff
@@ -36,3 +42,96 @@ pub(crate) fn create_anthropic(model_id: Option<&str>) -> Result<Arc<dyn Chat>,
 
     Ok(Arc::new(model))
 }
+
+pub(crate) async fn create_local() -> Result<Arc<dyn Chat>, anyhow::Error> {
+    let temp_dir = local_model_dir("Phi-3-mini-4k-instruct").await?;
+
+    download_hf_model_artifacts(
+        "microsoft/Phi-3-mini-4k-instruct",
+        None,
+        std::env::var("SPICE_HF_API_KEY").ok(),
+        vec![
+            "model-00001-of-00002.safetensors",
+            "model-00002-of-00002.safetensors",
+            "config.json",
+            "tokenizer.json",
+            "tokenizer_config.json",
+        ],
+        &temp_dir,
+    )
+    .await?;
+
+    let model_weights = [
+        temp_dir
+            .join("model-00001-of-00002.safetensors")
+            .to_str()
+            .unwrap_or_default()
+            .to_string(),
+        temp_dir
+            .join("model-00002-of-00002.safetensors")
+            .to_str()
+            .unwrap_or_default()
+            .to_string(),
+    ];
+
+    let model = create_local_model(
+        &model_weights,
+        temp_dir.join("config.json").to_str(),
+        temp_dir.join("tokenizer.json").to_str(),
+        temp_dir.join("tokenizer_config.json").to_str(),
+        None,
+    )
+    .map_err(anyhow::Error::from)?;
+    Ok(Arc::from(model))
+}
+
+/// Creates a directory for the specified model under `.spice/test_models`.
+pub async fn local_model_dir(model_name: &str) -> Result<PathBuf, anyhow::Error> {
```

Comment:
I don't think we need to move them to `.spice/test_models`? Loading them from `~/.cache/huggingface/...` should be simpler?

Reply:
Good point - this was added to make the test more tailored to the target use case and ensure this scenario (custom folder) is supported.